internal/vxproto/proto_test.go

```diff
@@ -1481,34 +1610,141 @@ func TestIMCSendToUnknownDestination(t *testing.T) {
+			t.Fatal("failed to send data packet to modude via imc", err)
+		}
+		err = moduleSocket.SendTextTo(ctx, dst, &Text{Data: testData, Name: textName})
+		if !errors.Is(err, d.expect) {
+			t.Fatal("failed to send text packet to modude via imc", err)
+		}
+		err = moduleSocket.SendFileTo(ctx, dst, &File{Data: testData, Name: fileName, Path: ""})
+		if !errors.Is(err, d.expect) {
+			t.Fatal("failed to send file packet to modude via imc", err)
+		}
+		err = moduleSocket.SendMsgTo(ctx, dst, &Msg{Data: testData, MType: msgType})
+		if !errors.Is(err, d.expect) {
+			t.Fatal("failed to send message packet to modude via imc", err)
+		}
+	}
+}
+
+func TestIMCTopicsAPI(t *testing.T) {
+	ctx := context.Background()
+	proto, err := getVXProto()
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer proto.Close(ctx)
+
+	moduleSocket := proto.NewModule("test", groupID)
+	if moduleSocket != nil {
+		if !proto.AddModule(moduleSocket) {
+			t.Fatal("failed to add Module Socket object")
+		}
+	} else {
+		t.Fatal("failed to initialize Module Socket object")
 	}
-	if err := moduleSocket.SendTextTo(ctx, dst, &Text{Data: testData, Name: textName}); err == nil {
-		t.Fatal("Failed send text packet to modude via imc:", err.Error())
+	defer func() {
+		if !proto.DelModule(moduleSocket) {
+			t.Fatal("failed to delete Module Socket object")
+		}
+	}()
+
+	imcToken := moduleSocket.GetIMCToken()
+	testData := []byte("test data message")
+	for idx := 0; idx < 10000; idx++ {
+		// On publisher side: should make topic token
+		topicName := fmt.Sprintf("test_topic%d", idx)
+		topicToken := moduleSocket.MakeIMCTopic(topicName, groupID)
+		if !strings.HasPrefix(topicToken, "ffff7777") {
+			t.Fatal("failed to prepare topic token")
+		}
+
+		// On publisher side: check is topic available
+		if tokenInfo := moduleSocket.GetIMCTopic(topicToken); tokenInfo != nil {
+			// This topic doesn't contain any subscribers and doesn't register in VXProto
+			t.Fatal("unregistered topic must return nil as an info")
+		}
+
+		// On publisher side: test sending packet to unavailable topic
+		err := moduleSocket.SendDataTo(ctx, topicToken, &Data{Data: testData})
+		if !errors.Is(err, ErrTopicUnreachable) {
+			// This call to send any packets to unavailable topic must raise the error
+			t.Fatal("send data call returned unexpected error", err)
+		}
+
+		// On subscriber side: subsribe to the topic and add an IMC token to the list
+		if !moduleSocket.SubscribeIMCToTopic(topicName, groupID, imcToken) {
+			// This topic was unregistered and must be free to subscribe on it
+			t.Fatal("unregistered topic must be available to subsribing")
+		}
+
+		// On publisher side: double check is topic available
+		if topicInfoR := moduleSocket.GetIMCTopic(topicToken); topicInfoR == nil {
+			// This topic must be registered in VXProto after previosly SubscribeIMC* call
+			t.Fatal("registered topic must return info about itself")
+		} else if topicInfoR.GetName() != topicName || topicInfoR.GetGroupID() != groupID {
```

Comment:
there is no need to use "else if" ad t.Fatal() breaks test execution so simple:
```
if {
}
if {
}
```
seems more readable

Reply:
I'm not on the same page with you because reviewer must remember end line for each branch